koswat/cost_report/summary/koswat_summary_location_matrix_builder.py

```diff
@@ -89,17 +89,14 @@ def get_reinforcement(
                 ground_level_surface=_reinforcement.new_ground_level_surface,
             )
 
-        _reinforcement_costs = [
-            next(iter(_reinforcement_dict.values()), None)
-            for _reinforcement_dict in locations_per_reinforcement
-        ]
-
-        return list(
-            map(
-                get_reinforcement,
-                filter(lambda x: x is not None, _reinforcement_costs),
-            )
-        )
+        # Get the reinforcement cost of the first location for each reinforcement (if present).
+        _reinforcement_costs = []
+        for _ref_loc_dict in locations_per_reinforcement:
+            if not _ref_loc_dict.values():
+                continue
+            _reinforcement_costs.append(list(_ref_loc_dict.values())[0])
```

Comment:
try not to repeat instantiation (thus computing times) of generators / iterators.